### DISCUSSION TITLE
Hey, This is an idea for a plugin.

### DISCUSSION THREAD

**Discussion by lsproule** [nvim-neorg/neorg#1073] [Ideas]
In order to make this a lot more like org mode. I would love to be able to in the header be able to specify a file. 

Then, in the 

@code javascript  
@end

it will write this to the file, so we can write the docs   and the file at the same  time
potentially
To  support multiple files and filetypes it may be interesting to add an additional  optional second argument to the code block

@code javascript vite.config.js
import defineConfig

export default {
    server:  {
        port:3000
    }
}
@end

[then since I am parsing this anyway. It shouldn't be hard to add this Repl ](https://github.com/pappasam/nvim-repl)

and just like in org mode we could have output directly below the code block
`
@code
x=5
print(f"{x = }")
@end

@output
x = 5
@end

**Comment by lsproule** [comment]
I realize this maybe off topic since it's an extension and not part of neorg itself . I just really would love these features, so if anyone would also like this and would like to help or have ideas on how to help implement any of them. I would love to hear more from you.

**Comment by max397574** [comment]
https://github.com/nvim-neorg/neorg/wiki/Tangling
